src/main/java/frc/robot/systems/KitBotShooterFSM.java

```diff
@@ -115,18 +143,41 @@ public boolean updateAutonomous(AutoFSMState autoState) {
 	 *        the robot is in autonomous mode.
 	 * @return FSM state for the next iteration
 	 */
-	private FSMState nextState(TeleopInput input) {
+	private ShooterFSMState nextState(TeleopInput input) {
+		if (input == null) {
+			return ShooterFSMState.IDLE_STOP;
+		}
 		switch (currentState) {
-			case START_STATE:
-				if (input != null) {
-					return FSMState.OTHER_STATE;
-				} else {
-					return FSMState.START_STATE;
+			case IDLE_STOP:
+				if (input.isOuttakeButtonPressed() && !input.isIntakeButtonPressed()) {
+					return ShooterFSMState.OUTTAKING;
+				}
+				if (input.isIntakeButtonPressed() && !hasNote()
+					&& !input.isOuttakeButtonPressed()) {
+					return ShooterFSMState.INTAKING;
+				}
+				if ((input.isIntakeButtonPressed() && hasNote())
+					|| !(input.isOuttakeButtonPressed() || input.isIntakeButtonPressed())
+					|| (input.isIntakeButtonPressed() && input.isOuttakeButtonPressed())) {
+					return ShooterFSMState.IDLE_STOP;
```

Comment:
Instead of having a whole separate if statement to return IDLE_STOP, I would just say you return IDLE_STOP without an if statement. So for each state, the first if statements check if the conditions are met to transition to a different state. And if they all fail, you just return IDLE_STOP

Reply:
do the same for each case in nextState